pkg/resource/stack/deployment.go

```diff
@@ -505,6 +505,17 @@ func DeserializePropertyValue(v interface{}, dec config.Decrypter) (resource.Pro
 						cachingCrypter.insert(prop.SecretValue(), plaintext, ciphertext)
 					}
 					return prop, nil
+				case resource.ResourceSig:
+					urn, ok := objmap["urn"].(string)
```

Comment:
Are we requiring that resource URNs are not secret?

Reply:
Yes - I'm happy with that restriction. I can't imagine what exactly it would mean for a URN to be a secret - we have to populate a full resources based on this.